### PR TITLE
Add Button to Milestone Notifications detail view

### PR DESCRIPTION
**Pull request by frosty**
This PR adds the ability for notifications to show a button in their detail view. Notification text blocks with a `meta` value of `"button": true` will be assigned the `button` kind, and rendered using the new `NoteBlockButtonTableViewCell`.

**Note**

The button currently takes you to the notifications day view for the specified site, but our system for routing those deep links currently doesn't respect the date range passed to it. We'll tweak that in a future PR.

**To test**

- You'll need to have the backend diff for Milestone Notifications applied to your sandbox and your simulator or device set up to use this for API calls.
- Ensure you have a Milestone Notification present in your notifications list.
- Enable the milestone notifications feature flag.
- Build and run and view the notification. Check that the **Take a deep dive button** is present.
- Check other notifications, and ensure that the button doesn't show up elsewhere.
- Ensure tests pass.
- Ensure that the button doesn't show up if the feature flag is disabled.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.swift [hunk 1]
```diff
@@ -860,6 +865,24 @@ private extension NotificationDetailsViewController {
...
+        }
+
+        cell.title = textBlock.text
+
+        if let linkRange = textBlock.ranges.map({ $0 as? LinkContentRange }).first,
+           let url = linkRange?.url {
+            cell.action = { [weak self] in
+                if UniversalLinkRouter.shared.canHandle(url: url) {
+                    UniversalLinkRouter.shared.handle(url: url, shouldTrack: false, source: self)
+                }
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.swift]: When I tap on the button there's like a half a second lag, then the app switches tabs, and then pushes to the stats view. 

What do you think about Reusing the `self.displayURL(url)` from the other actions which then just pushes to the stats view in the notifications tab?

File: WordPress/Classes/ViewRelated/Notifications/Views/NoteBlockButtonTableViewCell.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
+//
+//  NoteBlockButtonTableViewCell.swift
+//  WordPress
+//
+//  Created by James Frost on 19/02/2021.
+//  Copyright © 2021 WordPress. All rights reserved.
+//
+
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Notifications/Views/NoteBlockButtonTableViewCell.swift]:
```suggestion
```

File: WordPress/Classes/ViewRelated/Notifications/FormattableContent/NotificationTextContent.swift [hunk 3]
```diff
@@ -62,6 +63,13 @@ class NotificationTextContent: FormattableTextContent, FormattableMediaContent {
         if let firstMedia = media.first, (firstMedia.kind == .image || firstMedia.kind == .badge) {
             return .image
         }
+
+        if let meta = meta,
+           let buttonValue = meta[Constants.MetaKeys.Button] as? Bool,
+           buttonValue == true {
+            return .button
```
Comment (emilylaguna) [inline WordPress/Classes/ViewRelated/Notifications/FormattableContent/NotificationTextContent.swift:70]: What do you think about adding a `isViewMilestone` check here? I'm not 100% the button meta flag is exclusive to our notification.

File: WordPress/Classes/ViewRelated/Notifications/Views/NoteBlockButtonTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,40 @@
+//
+//  NoteBlockButtonTableViewCell.swift
```
Comment (yaelirub) [inline WordPress/Classes/ViewRelated/Notifications/Views/NoteBlockButtonTableViewCell.swift]: I think you forgot to remove these?

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Heya, this is working with the new notification. I left a few comments below.

When in dark mode there's a light background on the button section:

File: WordPress/Classes/ViewRelated/Notifications/Views/NoteBlockButtonTableViewCell.swift (hunk 4 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/Notifications/Views/NoteBlockButtonTableViewCell.swift]: D'oh, thank you! 🙂

File: WordPress/Classes/ViewRelated/Notifications/FormattableContent/NotificationTextContent.swift (hunk 3 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/Notifications/FormattableContent/NotificationTextContent.swift:70]: Sure!

File: WordPress/Classes/Models/Notifications/Notification.swift [hunk 5]
```diff
@@ -216,6 +216,10 @@ extension Notification {
         return block.isActionEnabled(id: commandId) && !block.isActionOn(id: commandId)
     }
 
+    var isViewMilestone: Bool {
+        return FeatureFlag.milestoneNotifications.enabled && type == "view_milestone"
```
Comment (Gio2018) [inline WordPress/Classes/Models/Notifications/Notification.swift:220]: nit: `return` is redundant, if you wish.

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.swift (hunk 1 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.swift]: Thanks, great suggestion! This seems to work really well – just looks like we'll need to tweak it to handle the date range 👍

**Comment by frosty** [comment]
@emilylaguna Ready for another look! 👀

**Review by emilylaguna — APPROVED** [review]
Nice work! 🚀

**Review by Gio2018 — CHANGES_REQUESTED** [review]
This looks really good @frosty !!
A minor comment that can safely be ignored in the code, and one issue that I found:

- There seems to be a strange "persistence" of the stats between My Site and Notifications. See the video below: the site selected in "My site" is different than the one of the notification


https://user-images.githubusercontent.com/34376330/108907188-678b2700-75e7-11eb-9717-bbed875619f7.mp4

File: WordPress/Classes/Models/Notifications/Notification.swift (hunk 5 above)
Comment (frosty) [reply WordPress/Classes/Models/Notifications/Notification.swift:220]: Hmm, I was kind of matching the style of the other one-line computed vars here – are you okay if we leave as-is?

**Review by Gio2018 — APPROVED** [review]

**Pull request merged**